com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs

```diff
@@ -126,23 +138,52 @@ static HDLightUI()
                 CED.Group((serialized, owner) =>
                     EditorGUILayout.HelpBox(s_Styles.unsupportedPresetPropertiesMessage, MessageType.Info)),
                 CED.Group((serialized, owner) => EditorGUILayout.Space()),
-                CED.FoldoutGroup(s_Styles.generalHeader, Expandable.General, k_ExpandedStatePreset, DrawGeneralContent),
-                CED.FoldoutGroup(s_Styles.emissionHeader, Expandable.Emission, k_ExpandedStatePreset, DrawEmissionContentForPreset),
+                CED.FoldoutGroup(s_Styles.generalHeader, Expandable.General, k_ExpandedStatePreset, CED.Group(
+                    (s, e) => DrawLightType(s, e, true),
+                    DrawGeneralContent)),
+                CED.FoldoutGroup(s_Styles.emissionHeader, Expandable.Emission, k_ExpandedStatePreset, CED.Group(DrawColor, DrawEmissionContent)),
                 CED.FoldoutGroup(s_Styles.shadowHeader, Expandable.Shadows, k_ExpandedStatePreset, DrawEnableShadowMapInternal)
             );
+
+            Inspector = CED.GroupWithPreset(normalInspector, presetInspector);
+
+            Type lightMappingType = typeof(Lightmapping);
+            var getLightingSettingsOrDefaultsFallbackInfo = lightMappingType.GetMethod("GetLightingSettingsOrDefaultsFallback", BindingFlags.Static | BindingFlags.NonPublic);
+            var getLightingSettingsOrDefaultsFallbackLambda = Expression.Lambda<Func<LightingSettings>>(Expression.Call(null, getLightingSettingsOrDefaultsFallbackInfo));
+            GetLightingSettingsOrDefaultsFallback = getLightingSettingsOrDefaultsFallbackLambda.Compile();
         }
 
-        static void DrawGeneralContent(SerializedHDLight serialized, Editor owner)
+        // This scope is here mainly to keep pointLightHDType isolated
+        public struct LightTypeEditionScope : IDisposable
+        {
+            EditorGUI.PropertyScope lightTypeScope;
+            EditorGUI.PropertyScope pointLightScope;
+
+            public LightTypeEditionScope(Rect rect, GUIContent label, SerializedHDLight serialized, bool isPreset)
+            {
+                lightTypeScope = new EditorGUI.PropertyScope(rect, label, serialized.settings.lightType);
+
+                // When editing a Light Preset, the HDAdditionalData, is not editable as is not shown on the inspector, therefore, all the properties
+                // That come from the HDAdditionalData are not editable, if we use the PropertyScope for those, as they are not editable this will block
+                // the edition of any property that came afterwards. So make sure that we do not use the PropertyScope if the editor is for a preset
+                pointLightScope = isPreset ? null : new EditorGUI.PropertyScope(rect, label, serialized.pointLightHDType);
+            }
+
+            void IDisposable.Dispose()
+            {
+                lightTypeScope.Dispose();
+                pointLightScope?.Dispose();
+            }
+        }
+
+        static bool s_LightTypeChanged = false;
```

Comment:
I'm not a fan of storing state in a static variable here, and using it in the next function call. I wonder if there's a better way to do this. To be fair, I don't really understand what's happening here.

Reply:
Changed ;)